ui/types/api/dialog.d.ts

```diff
@@ -1,12 +1,7 @@
 // Error on "quasar" import shown in IDE is normal, as we only have Components/Directives/Plugins types after the build step
 // The import will work correctly at runtime
 import { QInputProps, QOptionGroupProps } from "quasar";
-
-// Taken from https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#distributive-conditional-types
-type NonFunctionPropertyNames<T> = {
-  [K in keyof T]: T[K] extends Function ? never : K;
-}[keyof T];
-type NonFunctionProperties<T> = Pick<T, NonFunctionPropertyNames<T>>;
+import { InputHTMLAttributes, HTMLAttributes } from "vue";
```

Comment:
After your comment on extra stuff like `dataset` slipping in, I dug through and decided the best way is to use Vue's JSX typings. It only includes properties that are actually fillable via JSX/h(). Not all stuff is perfect, so some stuff like the specific input types that I made is still helpful and used. We can use types like these in more components as the time goes to improve the overall experience.